hal/src/electron/modem/mdm_hal.cpp

```diff
@@ -1137,6 +1137,17 @@ bool MDMParser::init(DevStatus* status)
         {
             bool continueInit = false;
             int newProf = UBLOX_SARA_UMNOPROF_SIM_SELECT;
+            // Hard code ATT for 05.12 firmware versions
+            if (strstr(_verExtended, "L0.0.00.00.05.12")) {
+                if (netProv == CELLULAR_NETPROV_KORE_ATT) {
+                    newProf = UBLOX_SARA_UMNOPROF_ATT;
+                    // continue on with init if we are trying to set SIM_SELECT a third time
+                    if (_resetFailureAttempts >= 2) {
```

Comment:
Isn't this exactly the same as https://github.com/particle-iot/device-os/pull/2318/files#diff-58aaafe528bfacd8e545300d32b1dcb9a353161cb8a10735e10951a83994cacaR1168 and we'll actually fall into that condition as well due to how this is structured.

Actually same thing in the Gen 3 PR: https://github.com/particle-iot/device-os/pull/2317/files#diff-a8e99f7ec884e049b049347a881b289b8edd5641af45e8d407cb97a78fdf7550R1010

Can we rearrange things a bit and do in both implementations perhaps?

1. `if CELLULAR_NETPROV_TWILIO`
2. `else`
2.1. `if 05.12 && CELLULAR_NETPROV_KORE_ATT`
3. Reset stuff

Reply:
Much neater 👍  . One other thing I noticed - and from what I am seeing, this may be attributed to subtle time differences - is that in _all_ runs with the new commit, the radio doesn't respond to the `UMNOPROF=2` command (or perhaps the response is not collected?). Checked that the modem is ON as read from `RI_UC` pin. 
And the situation has overall improved by adding an additional 200ms [here](https://github.com/particle-iot/device-os/pull/2318/files#diff-58aaafe528bfacd8e545300d32b1dcb9a353161cb8a10735e10951a83994cacaR1177) with every run
One such failing log:
```
    70.788 AT send      10 "AT+CFUN?\r\n"
    70.797 AT read  +   12 "\r\n+CFUN: 1\r\n"
    70.798 AT read OK    6 "\r\nOK\r\n"
    70.799 AT send      13 "AT+CFUN=0,0\r\n"
    70.974 AT read OK    6 "\r\nOK\r\n"
    70.975 AT read  +   14 "\r\n+CIEV: 7,0\r\n"
    71.975 AT send      15 "AT+UMNOPROF=2\r\n"
Modem State: 1
    81.976 AT send       4 "AT\r\n"

[ Modem::powerOn ] = = = = = = = = = = = = = =
Modem State: 1
    85.287 AT send       4 "AT\r\n"
Modem State: 1
    86.598 AT send       4 "AT\r\n"
Modem State: 1
    87.909 AT send       4 "AT\r\n"
Modem State: 1
    89.220 AT send       4 "AT\r\n"
Modem State: 1
    90.531 AT send       4 "AT\r\n"
Modem State: 1
    91.842 AT send       4 "AT\r\n"
Modem State: 1
    93.153 AT send       4 "AT\r\n"
Modem State: 1
    94.464 AT send       4 "AT\r\n"
Modem State: 1
    95.775 AT send       4 "AT\r\n"
Modem State: 1
    97.086 AT send       4 "AT\r\n"
Modem State: 1
    98.397 AT send       4 "AT\r\n"
Modem State: 1
    99.708 AT send       4 "AT\r\n"
Modem State: 1
   101.019 AT send       4 "AT\r\n"
Modem State: 1
   102.330 AT send       4 "AT\r\n"
Modem State: 1
   103.641 AT send       4 "AT\r\n"
Modem State: 1
   104.952 AT send       4 "AT\r\n"
Modem State: 1
   106.263 AT send       4 "AT\r\n"
Modem State: 1
   107.574 AT send       4 "AT\r\n"
Modem State: 1
   108.885 AT send       4 "AT\r\n"
Modem State: 1
   110.196 AT send       4 "AT\r\n"
Modem State: 1
   111.507 AT send       4 "AT\r\n"
Modem State: 1
   112.818 AT send       4 "AT\r\n"
Modem State: 1
   114.129 AT send       4 "AT\r\n"
Modem State: 1
   115.440 AT send       4 "AT\r\n"
Modem State: 1
   116.751 AT send       4 "AT\r\n"
[ Modem reset ]
Modem State: 1
   128.062 AT send       4 "AT\r\n"
Modem State: 1
   129.373 AT send       4 "AT\r\n"
Modem State: 1
   130.684 AT send       4 "AT\r\n"
Modem State: 1
   131.995 AT send       4 "AT\r\n"
Modem State: 1
   133.306 AT send       4 "AT\r\n"
   133.745 AT read UNK   3 "AT\r"
   133.746 AT read OK    6 "\r\nOK\r\n"
   133.746 AT send       9 "AT+CGMM\r\n"
   133.760 AT read UNK   8 "AT+CGMM\r"
   133.761 AT read UNK  18 "\r\nSARA-R410M-02B\r\n"
   133.763 AT read OK    6 "\r\nOK\r\n"
   133.787 AT read  +   11 "\r\n+PACSP0\r\n"
   133.967 AT send       6 "ATE0\r\n"
   133.971 AT read UNK   5 "ATE0\r"
   133.972 AT read OK    6 "\r\nOK\r\n"
   133.972 AT send      11 "AT+CMEE=2\r\n"
   133.980 AT read OK    6 "\r\nOK\r\n"
   133.980 AT send      19 "AT+CMER=1,0,0,2,1\r\n"
   133.989 AT read OK    6 "\r\nOK\r\n"
   133.989 AT send      15 "AT+IPR=115200\r\n"
   134.000 AT read OK    9 "\r\n\r\r\nOK\r\n"
   134.100 AT send      10 "AT+CPIN?\r\n"
   134.105 AT read  +   14 "\r\n+CIEV: 7,1\r\n"
   134.107 AT read  +   16 "\r\n+CPIN: READY\r\n"
   134.109 AT read OK    6 "\r\nOK\r\n"

[ Modem::init (attempt 2/8) ] = = = = = = = = = =
   134.366 AT send       9 "AT+CGSN\r\n"
   134.373 AT read UNK  19 "\r\n352753090703578\r\n"
   134.374 AT read OK    6 "\r\nOK\r\n"
   134.375 AT send       9 "AT+CGMI\r\n"
   134.385 AT read UNK  10 "\r\nu-blox\r\n"
   134.386 AT read OK    6 "\r\nOK\r\n"
   134.386 AT send       9 "AT+CGMR\r\n"
   134.398 AT read UNK  43 "\r\nL0.0.00.00.05.12 [Oct 08 2020 15:00:01]\r\n"
   134.400 AT read OK    6 "\r\nOK\r\n"
   134.402 AT send       6 "ATI9\r\n"
   134.412 AT read UNK  28 "\r\nL0.0.00.00.05.12,A.02.19\r\n"
   134.414 AT read OK    6 "\r\nOK\r\n"
   134.415 AT send       9 "AT+CCID\r\n"
   134.428 AT read  +   31 "\r\n+CCID: 89014103259631211821\r\n"
   134.430 AT read OK    6 "\r\nOK\r\n"
   134.431 AT send      11 "AT+CMGF=1\r\n"
   134.440 AT read OK    6 "\r\nOK\r\n"
   134.440 AT send      13 "AT+CNMI=2,1\r\n"
   134.449 AT read OK    6 "\r\nOK\r\n"
   134.449 AT send       9 "AT+CIMI\r\n"
   134.456 AT read UNK  19 "\r\n310410963121182\r\n"
   134.457 AT read OK    6 "\r\nOK\r\n"
   134.458 AT send      13 "AT+COPS=3,2\r\n"
   134.469 AT read OK    6 "\r\nOK\r\n"
   134.469 AT send      14 "AT+UMNOPROF?\r\n"
   134.480 AT read  +   16 "\r\n+UMNOPROF: 2\r\n"
   134.481 AT read OK    6 "\r\nOK\r\n"
```